libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -855,19 +856,41 @@ void AC_PosControl::input_vel_accel_z(float &vel, const float accel, bool ignore
 
 /// set_pos_target_z_from_climb_rate_cm - adjusts target up or down using a commanded climb rate in cm/s
 ///     using the default position control kinematic path.
+///     The zero target altitude is varied to follow pos_offset_z
+void AC_PosControl::set_pos_target_z_from_climb_rate_cm(float vel)
+{
+    // remove terrain offsets for flat earth assumption
+    _pos_target.z -= _pos_offset_z;
+    _vel_desired.z -= _vel_offset_z;
+    _accel_desired.z -= _accel_offset_z;
+
+    float vel_temp = vel;
+    input_vel_accel_z(vel_temp, 0, false);
+
+    // update the vertical position, velocity and acceleration offsets
+    update_pos_offset_z(_pos_offset_target_z);
+
+    // add terrain offsets
+    _pos_target.z += _pos_offset_z;
+    _vel_desired.z += _vel_offset_z;
+    _accel_desired.z += _accel_offset_z;
+}
+
+/// land_at_climb_rate_cm - adjusts target up or down using a commanded climb rate in cm/s
+///     using the default position control kinematic path.
 ///     ignore_descent_limit turns off output saturation handling to aid in landing detection. ignore_descent_limit should be true unless landing.
-void AC_PosControl::set_pos_target_z_from_climb_rate_cm(const float vel, bool ignore_descent_limit)
+void AC_PosControl::land_at_climb_rate_cm(float vel, bool ignore_descent_limit)
 {
-    float vel2 = vel;
-    input_vel_accel_z(vel2, 0, ignore_descent_limit);
+    float vel_temp = vel;
+    input_vel_accel_z(vel_temp, 0, ignore_descent_limit);
```

Comment:
```suggestion
    input_vel_accel_z(vel, 0, ignore_descent_limit);
```
Without the const we don't need this extra var.

Reply:
Thanks!